File: RoomExample/app/src/main/java/com/madonasyombua/roomexample/room/dao/ItemDao.java
package com.madonasyombua.roomexample.room.dao;


import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.madonasyombua.roomexample.room.entity.Item;

import java.util.List;

/**
 * @author madona syombua
 *
 * Dao , database accessible object for the item
 * Query - from the db
 * Insert - to the db
 * Delete - from the db
 * Update - the db
 */


@Dao
public interface ItemDao {


    @Query("SELECT * FROM items ORDER BY id")
    LiveData<List<Item>> getAllItems();

    @Insert
    void insertItem(Item item);


    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateItems(Item item);

    @Delete
    void deleteItem(Item item);


}

File: RoomExampleKotlin/app/src/main/java/com/madonasyombua/roomexamplekotlin/room/dao/ItemDao.kt
package com.madonasyombua.roomexamplekotlin.room.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.madonasyombua.roomexamplekotlin.room.entity.Item

/**
 * @author madona syombua
 *
 * Dao , database accessible object for the item
 * Query - from the db
 * Insert - to the db
 * Delete - from the db
 * Update - the db
 */


@Dao
interface ItemDao {


    @get:Query("SELECT * FROM items ORDER BY id")
    val allItems: LiveData<List<Item>>

    @Insert
    fun insertItem(item: Item)


    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateItems(item: Item)

    @Delete
    fun deleteItem(item: Item)


}
